Update config.cfg to include z1 and z2 instead of extruder which is used from toolboard

nickidotdk opened the pull request:
When configuring a VCore3 or 4 the configuration is missing configuration for z1 and z2. In the configurator it states that when using this board a toolboard is required. Going through the steps and choosing the motors for z1 and z2 works fine, but at the last step it fails since there were no configurations defined for z1 and z2.  This fixes that issue.

nickidotdk committed:
Update config.cfg to include z1 and z2 instead of extruder which is used from toolboard

When configuring a VCore3 or 4 the configuration is missing configuration for z1 and z2. In the configurator it states that when using this board a toolboard is required. Going through the steps and choosing the motors for z1 and z2 works fine, but at the last step it fails since there were no configurations defined for z1 and z2. 
This fixes that issue.

miklschmidt commented:
This change would make the board unusable on printers without a toolboard. However i think the actual bug here is that you managed to get through the hardware configuration without assigning motor slots to the undefined axes. Sounds like this isn't an issue anymore though?

In 2.0 we used a separate config-3z.cfg to make 5 driver boards compatible with printers that need 6 steppers.

nickidotdk commented:
I did assign the motor slots, and then the very last page - the confirmation page, fails due to the missing motor section in the config file that it loads data from when injecting into the config file.

nickidotdk commented:
But I see your point regarding not working for non toolboard printers. The configurator could use individual confog files for the different printers.

nickidotdk commented:
I will only use this locally for myself since this introduces unwanted behaviour for non vcore 3.x/4 printers.

miklschmidt commented:
@nickidotdk I'm gonna reopen this as there's clearly still a bug somewhere that should be addressed, and we should probably add a config-3z.cfg with the contents of this PR (instead of modifying the original). So what needs to be done is:

1) Add config-3z.cfg with the contents of this PR's config.cfg.
2) Add `"extruderlessConfig": "config-3z.cfg"` to `board-definition.json`
3) Before merging, reproduce and fix the issue in the configurator (https://github.com/Rat-OS/RatOS-configurator).

To clarify point 3:
- When motor slots are assigned, the pins from `board-definition.json` is used to generate the board_pin alias map, in place of the values from `config.cfg` (or else it would not work for any includes/configs that use the board pin alias).
- Fixing this issue involves finding out what is generating the error and why. The user should not be able to proceed from the hardware configuration step without resolving any driver slot conflicts or errors.
- Provided that all "printer rails" have a motor slot assignment, there should be no error about stepper pins further in the process, regardless of what's in config.cfg or if a `extruderlessConfig` is present.

nickidotdk commented:
@miklschmidt I have made the changes. I was not sure how to include both changes in the same PR, so they are seperate. 
https://github.com/Rat-OS/RatOS-configuration/pull/295
https://github.com/Rat-OS/RatOS-configuration/pull/294
I don't have a linux box to test this on. Could you help with that?

Also I wanted to delete the PRs made by Calcodk - I was logged into the wrong account when doing the changes. I am not used to working directly in Github :)

Best regards.

nickidotdk commented:
@miklschmidt It is working as intended. I have tested it by manually adding config-3z.cfg and changing board-definition.json in my RatOS installation on the M5P. Now everything is working. I also changed from CAN to Usb and added a Beacon. Waow what a great job you guys did with this!!!! <3

miklschmidt commented:
@nickidotdk Thank you for your contribution and thanks for the kind words!! :)
